src/playlist-loader.js

```diff
@@ -79,11 +79,59 @@ export const updateMaster = (master, media) => {
 
   // consider the playlist unchanged if the number of segments is equal, the media
   // sequence number is unchanged, and this playlist hasn't become the end of the playlist
-  if (playlist.segments &&
-      media.segments &&
-      playlist.segments.length === media.segments.length &&
-      playlist.endList === media.endList &&
-      playlist.mediaSequence === media.mediaSequence) {
+  let unchanged = playlist.segments &&
+    media.segments &&
+    playlist.segments.length === media.segments.length &&
+    playlist.endList === media.endList &&
+    playlist.mediaSequence === media.mediaSequence;
+
+  // for dash mediaSequence and segment lengths will often match as
+  // mediaSequence is almost always 1 and the number of segments generated for a
+  // given time is often the same. So we need to make sure that the underlying segments are
+  // different.
+  if (dash && unchanged) {
+    const oldSidx = playlist.sidx;
+    const newSidx = media.sidx;
+
+    // if sidx changed then the playlists are different.
+    if (oldSidx && newSidx && (oldSidx.offset !== newSidx.offset || oldSidx.length !== newSidx.length)) {
```

Comment:
Now dash playlists can also be listed as changing in cases where:
1. sidx does not match
2. playlist segment uris don't match up
3. playlist segment byterange uris don't match up.